accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,262 @@
+# Dart Constructor Tearoffs Feature FAQ
+
+Author: lrn@google.com<br>Version: 1.0
+
+This is a short summary of the _Constructor Tearoffs_ feature. This document is not intended as a specification, look at the feature specification for that. Instead it hopes to be a brief *introduction* to the feature set that we intend to release, and to answer some of the questions that it is hard to find short answers for in the specification.
+
+## What are the new features?
+
+In short:
+
+* Named constructor tear-off (`C.name` is a valid expression).
+* Named unnamed constructor (`C.new` is a constructor name, refers to the same constructor as "unnamed" `C` constructor).
+* Function value instantiation (you can instantiate function *values*, not just tear-offs).
+* Explicit instantiation (`List<int>` and `Future.then<int>` are valid type- and function-expressions).
+
+## Named constructor tear-off
+
+If *C* refers to a class, and *C* has a constructor *C*.*name*, then <code>*C*.*name*</code> can now be an expression which evaluates to a function with the same function signature as the constructor, and which, when called, does the same thing as the constructor.
+
+Example:
+
+```dart
+// Has type: DateTime Function(int, [int, int, int, int, int, int, int])
+var makeUtcDate = DateTime.utc; 
+```
+
+**Q:** What if the class is generic?
+
+**A:** Then the function is also generic, with the same type arguments as the class.
```

Comment:
I usually use `type arguments` to mean `actual type arguments` and `type parameters` to mean `formal type parameters`. In this case (where the reader can't be assumed to know the habits of the Dart specification documents) it would perhaps be best to use 'formal type parameters', such that anyone who has had a compiler course would get a rather firm hint about the intended meaning.